src/lib/y2storage/storage_manager.rb

```diff
@@ -27,76 +27,131 @@
 
 module Y2Storage
   #
-  # Singleton class for the libstorage object.
-  #
-  # You can simply use StorageManager.instance to use it and create a
-  # libstorage instance if there isn't one yet; this will use common default
-  # parameters for creating it.
-  #
-  # If you need special parameters for creating it, you can use
-  # StorageManager.create_instance with a custom ::Storage::Environment
-  #
-  # By default (unless disabled in the ::Storage::Environment), creating the
-  # instance will also start hardware probing. This is why there is an alias
-  # name StorageManager.start_probing to make this explicit.
+  # Singleton class to provide access to the libstorage object and associated
+  # information.
   #
   class StorageManager
     include Yast::Logger
+    extend Forwardable
+
+    # Libstorage object
+    #
+    # Calls to #probed, #staging, #environment and #arch are forwarded to this
+    # object.
+    #
+    # @return [Storage::Storage]
+    attr_reader :storage
+
+    # Revision of the staging devicegraph.
+    #
+    # Zero means no modification (staging looks like probed). Incremented every
+    # time the staging devicegraph is re-assigned.
+    # @see #copy_to_staging
+    #
+    # @return [Fixnum]
+    attr_reader :staging_revision
+
+    def_delegators :@storage, :probed, :staging, :environment, :arch
+
+    def initialize(storage_environment)
+      @storage = Storage::Storage.new(storage_environment)
+      @staging_revision = 0
+    end
+
+    # Sets the devicegraph as the staging one, updating all the associated
+    # information like #staging_revision
+    #
+    # @param [Storage::Devicegraph] devicegraph to copy
+    def copy_to_staging(devicegraph)
+      devicegraph.copy(storage.staging)
+      update_staging_revision
+    end
+
+    # Increments #staging_revision
+    #
+    # To be called explicitly if the staging devicegraph is modified without
+    # using #copy_to_staging
+    def update_staging_revision
+      @staging_revision += 1
+    end
+
     #
     # Class methods
     #
     class << self
-      # Return the singleton for the libstorage object. This will create one
-      # for the first call, which will also trigger hardware probing.
+      # Returns the singleton instance.
+      #
+      # In the first call, it will create a libstorage instance (using common
+      # defaults) if there isn't one yet. That means that, by default, the first
+      # call will also trigger hardware probing.
+      #
+      # @see .create_instance if you need special parameters for creating the
+      #   libstorage instance
+      # @see .create_test_instance if you just need to create an instance
+      #   without hardware probing
+      # @see .fake_from_yaml for easy mocking
       #
-      # @return [::Storage::Storage] libstorage object
+      # @return [StorageManager]
       #
       def instance
         create_instance unless @instance
```

Comment:
Well, we already had a lot of discussions about that when the class was originally created. I don't want to open that box again. :-)